File: multipaz-compose/src/commonMain/kotlin/org/multipaz/compose/cards/CardInfo.kt
package org.multipaz.compose.cards

import androidx.compose.ui.graphics.ImageBitmap

/**
 * Interface for information about a card to be displayed in a list.
 *
 * @property identifier a unique identifier for this card.
 * @property cardArt an image that represents this card to the user in the UI.
 * @property badges a list of badges to display on the card.
 */
interface CardInfo {
    val identifier: String
    val cardArt: ImageBitmap
    val badges: List<CardBadge>
}

File: multipaz-swiftui/src/iosMain/swift/CardInfo.swift
import UIKit

/// An interface for information about a card to be displayed in a list.
public protocol CardInfo {
    /// Unique identifier for the card.
    var identifier: String { get }
    /// Card art for the card.
    var cardArt: UIImage { get }
    /// Badges for the card.
    var badges: [CardBadge] { get }
}
